rs_server/s3_storage_handler/s3_storage_handler.py

```
@@ -17,243 +24,719 @@
+                    self.logger.error("This clent already exists")
+                else:
+                    self.logger.error("Unexpected error")
+                return None
+
+        return s3_client
+
+    def connect_s3(self):
+        """Establishe a connection to the S3 service.
+
+        Returns:
+            bool: True if the connection is successful, False otherwise.
+        """
+        if self.s3_client is None:
+            self.s3_client = self.__get_s3_client(
+                self.access_key_id,
+                self.secret_access_key,
+                self.endpoint_url,
+                self.region_name,
+            )
+        if self.s3_client is None:
+            return False
+        return True
+
+    def disconnect_s3(self):
+        """Close the connection to the S3 service."""
+        if self.s3_client is not None:
+            self.s3_client.close()
+        self.s3_client = None
+
+    # delete a file from s3 by using the s3 handler
+    def delete_file_from_s3(self, bucket, s3_obj):
+        """Delete a file from S3.
+
+        Args:
+            bucket (str): The S3 bucket name.
+            s3_obj (str): The S3 object key.
+
+        Returns:
+            bool: True if the file is successfully deleted, False otherwise.
+        """
+        if self.s3_client is None or bucket is None or s3_obj is None:
+            print("Input error for deleting the file")
+            return False
+        try:
+            with self.s3_client_mutex:
+                self.logger.debug("{%s} | {%s}", bucket, s3_obj)
+                self.logger.info("Delete file s3://{%s}/{%s}", bucket, s3_obj)
+                self.s3_client.delete_object(Bucket=bucket, Key=s3_obj)
+        except ClientError as e:
+            tb = traceback.format_exc()
+            self.logger.error("Failed to delete file s3://{%s}/{%s}", bucket, s3_obj)
+            self.logger.error("Exception: {%s} | {%s}", e, tb)
+            return False
+        return True
```

Here and in other places, also raise Exceptions instead of returning booleans. I've done it here: https://github.com/RS-PYTHON/rs-server/commit/974fe8ba79c6fc378733982946230feae8509a94

---

Merged with your branch